client/dashboard/sites/actions.tsx

```diff
@@ -49,6 +49,15 @@ export function getActions( router: AnyRouter ): Action< Site >[] {
 			},
 			isEligible: ( item: Site ) => canManageSite( item ),
 		},
+		{
+			id: 'jetpack-cloud',
+			label: __( 'Jetpack Cloud ↗' ),
+			callback: ( sites: Site[] ) => {
+				const site = sites[ 0 ];
+				window.open( `https://cloud.jetpack.com/landing/${ site.slug }` );
+			},
+			isEligible: ( item: Site ) => isSelfHostedJetpackConnected( item ),
```

Comment:
The previous logic had a `! canManageSite( item )` as well, but I didn't think it was necessary. `isSelfHostedJetpackConnected( item )` seemed enough.